include/deal.II/lac/block_linear_operator.h

```diff
@@ -77,7 +242,7 @@ block_operator(const std::array<std::array<LinearOperator<typename Range::BlockT
     v.reinit(m);
 
     // And reinitialize every individual block with reinit_range_vectors:
-    for (unsigned int i = 0; i < m; ++i)
+    for (size_t i = 0; i < m; ++i)
```

Comment:
Fair enough. We've often introduced a local typedef `size_type` in classes to denote this sort of type. You could here typedef it to
```
  typedef std::array<...>::size_type size_type;
```
to ensure that it always matches.